mysql/awsmysql/awsmysql.go

```diff
@@ -135,12 +133,12 @@ func (uo *URLOpener) OpenMySQLURL(ctx context.Context, u *url.URL) (*sql.DB, err
 
 type connector struct {
 	traceOpts []otelsql.Option
+	provider  CertPoolProvider
```

Comment:
Nit: in future, if you want to do refactorings like this, do them in a separate PR. When you change things that have nothing to do with the bug you're fixing, it makes the review harder. In this case you just reordered some fields in the struct, not sure why.